lib/schema/validate.js

```diff
@@ -17,6 +17,27 @@ const buildEnumMap = (enums = []) =>
     return acc;
   }, {});
 
+const resolveStringList = (string, enums) => {
+  // Quick check if the entire string is a valid enum, return it as an array
+  if (enums.includes(string)) return [string];
+
+  // Split the string on commas not followed by a space
```

Comment:
This need to do resolution by types instead. 

* If the underlying type for the enumeration is Edm.String, then it's a string value and should not be split.
* If it's anything else, it should be split on commas and treated as a list, if that's what it is.